src/model.coffee

```diff
@@ -169,6 +169,24 @@ class Model extends Backbone.Model
       )
       .promise(options.returnValues.jqXhr)
 
+  destroy: (options = {}) ->
+    cleanUpAssociatedReferences = =>
+      for collectionName, collection of @storageManager.collections
+        for reference, modelName of collection.modelKlass.associations
+          associationKey = collection.modelKlass.associationDetails(reference).key
+          if _.isArray(modelName) && _.find(modelName, (m) => inflection.singularize(m) == @className())
+            collection.storage.each (model) =>
+              model.set(associationKey, _.without(model.get(associationKey), @id))
+          else if !_.isArray(modelName) && inflection.singularize(modelName) == @className()
+            collection.storage.each (model) =>
```

Comment:
Also, if you want to make this more Underscore like, you can do

`collection.storage.invoke 'unset', associationKey`